src/type_traits/complex_help.hpp

```diff
@@ -26,6 +26,24 @@ using IsComplex = std::enable_if_t<implIsComplex<T>::value, bool>;
 template<typename T>
 using IsReal = std::enable_if_t<std::is_floating_point<T>::value, bool>;
 
+template <typename T, typename = bool>
+struct RealAlias_impl {};
```

Comment:
I believe there are more restrictions on T but revisit when taking care of #3378